mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -227,6 +236,25 @@ class NaiveBayesModel private[ml] (
 
   @Since("1.6.0")
   override def write: MLWriter = new NaiveBayesModel.NaiveBayesModelWriter(this)
+
+  private var trainingSummary: Option[NaiveBayesSummary] = None
+
+  private[classification] def setSummary(summary: NaiveBayesSummary): this.type = {
+    this.trainingSummary = Some(summary)
+    this
+  }
+
+  /**
+   * Gets summary of model on training set. An exception is thrown if `trainingSummary == None`.
+   */
+  @Since("2.0.0")
+  def summary: NaiveBayesSummary = trainingSummary match {
```

Comment:
Do we need to add this class? It doesn't provide extra summary statistics than the model itself.